core/embed/extmod/modtrezorcrypto/modtrezorcrypto-tropic.h

```diff
@@ -0,0 +1,282 @@
+
+  vstr_t result = {0};
+  vstr_init_len(&result, message_b.len);
+
+  memcpy(result.buf, msg_in, message_b.len);
+  result.len = strlen(result.buf);
+
+  return mp_obj_new_str_from_vstr(&mp_type_str, &result);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(mod_trezorcrypto_tropic_ping_obj,
+                                 mod_trezorcrypto_tropic_ping);
+
+/// def get_certificate() -> bytes:
+///     """
+///     Return the chip's certificate.
+///     """
+STATIC mp_obj_t mod_trezorcrypto_tropic_get_certificate() {
+  lt_handle_t handle = {0};
+  lt_ret_t ret = LT_FAIL;
+
+  tropic_init(&handle);
+
+  uint8_t X509_cert[512] = {0};
+  ret = lt_get_info_cert(&handle, X509_cert, 512);
+  if (ret != LT_OK) {
+    tropic_deinit(&handle);
+    mp_raise_msg(&mp_type_TropicError, "lt_get_info_cert failed.");
+  }
+
+  tropic_deinit(&handle);
+
+  vstr_t vstr = {0};
+  vstr_init_len(&vstr, 512);
+
+  memcpy(vstr.buf, X509_cert, 512);
+
+  return mp_obj_new_str_from_vstr(&mp_type_bytes, &vstr);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_0(mod_trezorcrypto_tropic_get_certificate_obj,
+                                 mod_trezorcrypto_tropic_get_certificate);
+
+/// def key_generate(
+///     key_index: int,
+/// ) -> None:
+///     """
+///     Generate ECC key in the device's ECC key slot.
+///     """
+STATIC mp_obj_t mod_trezorcrypto_tropic_key_generate(mp_obj_t key_index) {
+  mp_int_t idx = mp_obj_get_int(key_index);
+  if (idx < 0 || idx >= ECC_SLOT_COUNT) {
+    mp_raise_ValueError("Invalid index.");
+  }
+
+  lt_handle_t handle = {0};
+  lt_ret_t ret = LT_FAIL;
+
+  tropic_init(&handle);
+
+  ret = lt_ecc_key_generate(&handle, idx, CURVE_ED25519);
+  if (ret != LT_OK) {
+    tropic_deinit(&handle);
+    mp_raise_msg(&mp_type_TropicError, "lt_ecc_key_generate failed.");
+  }
+
+  tropic_deinit(&handle);
+
+  return mp_const_none;
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(mod_trezorcrypto_tropic_key_generate_obj,
+                                 mod_trezorcrypto_tropic_key_generate);
+
+/// def random_get(
+///     len: int,
+/// ) -> bytes:
+///     """
+///     Get number of random bytes.
+///     """
+STATIC mp_obj_t mod_trezorcrypto_tropic_random_get(mp_obj_t len) {
+  mp_int_t len_rand = mp_obj_get_int(len);
+  if (len_rand < 0 || len_rand > RANDOM_VALUE_GET_LEN_MAX) {
+    mp_raise_ValueError("Invalid length.");
+  }
+
+  lt_handle_t handle = {0};
+  lt_ret_t ret = LT_FAIL;
+
+  tropic_init(&handle);
+
+  uint8_t buff[RANDOM_VALUE_GET_LEN_MAX] = {0};
+  ret = lt_random_get(&handle, buff, len_rand);
+  if (ret != LT_OK) {
+    tropic_deinit(&handle);
+    mp_raise_msg(&mp_type_TropicError, "lt_random_get failed.");
+  }
+
+  tropic_deinit(&handle);
+
+  vstr_t vstr = {0};
+  vstr_init_len(&vstr, len_rand);
+
+  memcpy(vstr.buf, buff, len_rand);
+
+  return mp_obj_new_str_from_vstr(&mp_type_bytes, &vstr);
+}
+STATIC MP_DEFINE_CONST_FUN_OBJ_1(mod_trezorcrypto_tropic_random_get_obj,
+                                 mod_trezorcrypto_tropic_random_get);
+
```

Comment:
I'm not sure if we need this method. If we want to generate randomness in python, we would like to combine entropy from Tropic, Optiga and MCU, so we rather need to integrate this function [here](https://github.com/trezor/trezor-firmware/blob/ibz/20240805-libtropic/core/embed/extmod/modtrezorcrypto/modtrezorcrypto-random.h#L47).

Reply:
Removed it https://github.com/trezor/trezor-firmware/pull/4172/commits/0c33e899df511d6d4f3535c92c9a99cde2820fb8#diff-e1a3e03160d11080fcca39f3a8a9205fde724d8cd7d61c8505f6f41c9a95b04fL183